### PR TITLE
OpenSigning output

### PR DESCRIPTION
**Pull request by git-gud-1**
This commit formats log statements and dampens how often they are sent to the returner and removes  unused imported modules.

### CODE REVIEW

File: hubblestack/utils/signing.py [hunk 1]
```diff
@@ -124,10 +167,14 @@ def split_certs(fh):
...
                     yield ossl.load_certificate(ossl.FILETYPE_PEM, ret)
-                except Exception as exception_object:
-                    log.debug('decoding item as certificate failed: %s; trying as PEM encoded private key',
-                        exception_object)
+                except Exception as E:
+                    status = STATUS.UNKNOWN
+                    if check_verif_timestamp(fh) == True:
+                        log_level = log.error
+                    msg = '{}: | file: "{}" | cert decoding status: {} | attempting as PEM encoded private key'
+                    log_level(msg.format(short_fname, fh.name, status, digest, code, depth, message))
```
Comment (jettero) [inline hubblestack/utils/signing.py]: Avoid using `msg.format`; log.debug already issues a format, but using the older style (for various reasons). Please change to `log_level('%s: | file: "%s" | cert decoding status: %s | attempting as PEM encoded private key', short_fname, fh.name, ...)`

File: hubblestack/utils/signing.py [hunk 2]
```diff
@@ -124,10 +167,14 @@ def split_certs(fh):
             if line.startswith('----'):
                 ret = ret.encode()
                 try:
+                    log_level = log.debug
                     yield ossl.load_certificate(ossl.FILETYPE_PEM, ret)
-                except Exception as exception_object:
-                    log.debug('decoding item as certificate failed: %s; trying as PEM encoded private key',
-                        exception_object)
+                except Exception as E:
```
Comment (jettero) [inline hubblestack/utils/signing.py]: pylint will reject this on the variable name. Please rename to exception_object or similar.

File: hubblestack/utils/signing.py [hunk 3]
```diff
@@ -170,7 +232,7 @@ def authenticate_cert(self):
     def __init__(self, public_crt, ca_crt):
         try:
             import hubblestack.pre_packaged_certificates as HPPC
-            # iff we have hardcoded certs then we're meant to ignore any other
```
Comment (jettero) [inline hubblestack/utils/signing.py]: not a misspelling... iff means: if and only if

File: hubblestack/utils/signing.py [hunk 4]
```diff
@@ -229,34 +307,44 @@ def __init__(self, public_crt, ca_crt):
...
+                    # X509_V_ERR_CERT_HAS_EXPIRED                  10
+                    # X509_V_ERR_CRL_HAS_EXPIRED                   12
+                    status = STATUS.FAIL
+                # log at either log.error or log.critical according to the error code
+                if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
+                    if status == STATUS.FAIL: log_level = log.critical
+                    elif status == STATUS.UNKNOWN: log_level = log.error
+                    else: log_level = log.splunk
+                msg = 'public cert | file: "{}" | status: {} | digest: "{}" | X509 error code: {} | depth: {} | message: "{}"'
+                log_level(msg.format(str_public, status, digest, code, depth, message))
```
Comment (jettero) [inline hubblestack/utils/signing.py]: avoid new style msg.format()

File: hubblestack/utils/signing.py [hunk 5]
```diff
@@ -352,44 +440,59 @@ def sign_target(fname, ofname, private_key='private.key', **kwargs): # pylint: d
         args['algorithm'] = utils.Prehashed(chosen_hash)
     sig = first_key.sign(**args)
     with open(ofname, 'w') as fh:
-        log.debug('writing signature of %s to %s', os.path.abspath(fname), os.path.abspath(ofname))
+        log.debug('writing signature of {} to {}'.format(os.path.abspath(fname), os.path.abspath(ofname)))
```
Comment (jettero) [inline hubblestack/utils/signing.py]: please use log.debug's internall formatting

File: hubblestack/utils/signing.py [hunk 6]
```diff
@@ -352,44 +440,59 @@ def sign_target(fname, ofname, private_key='private.key', **kwargs): # pylint: d
...
+        sha256sum = hash_target(fname)
+        msg = '{} | file "{}" | status: {} | sha256sum: "{}" | public cert fingerprint and requester: "{}"'
         pubkey = crt.get_pubkey().to_cryptography_key()
         if isinstance(pubkey, rsa.RSAPublicKey):
             args['padding'] = padding.PSS( mgf=padding.MGF1(hashes.SHA256()),
                 salt_length=padding.PSS.MAX_LENGTH)
             args['algorithm'] = utils.Prehashed(chosen_hash)
         try:
             pubkey.verify(**args)
+            log_level(msg.format(short_fname, fname, status, sha256sum, txt ))
```
Comment (jettero) [inline hubblestack/utils/signing.py]: avoid new style formatting

File: hubblestack/utils/signing.py [hunk 7]
```diff
@@ -352,44 +440,59 @@ def sign_target(fname, ofname, private_key='private.key', **kwargs): # pylint: d
...
             return status
         except InvalidSignature:
+            status = STATUS.FAIL
+            if check_verif_timestamp(fname) == True:
+                log_level = log.critical
+            log_level(msg.format(short_fname, fname, status, sha256sum, txt))
```
Comment (jettero) [inline hubblestack/utils/signing.py]: new style formatting

File: hubblestack/utils/signing.py [hunk 8]
```diff
@@ -417,7 +521,8 @@ def verify_files(targets, mfname='MANIFEST', sfname='SIGNATURE', public_crt='pub
 
         return a mapping from the input target list to the status values (a dict of filename: status)
     """
-    log.debug("verify_files(%s, mfname=%s, sfname=%s, public_crt=%s, ca_crt=%s", targets, mfname, sfname, public_crt, ca_crt)
+    msg = "verifying: files: {} | mfname: {} | sfname: {} | public_crt: {}| ca_crt: {}"
+    log.debug(msg.format(targets, mfname, sfname, public_crt, ca_crt))
```
Comment (jettero) [inline hubblestack/utils/signing.py]: void new style formatting.

File: hubblestack/utils/signing.py [hunk 9]
```diff
@@ -503,8 +625,8 @@ def inner(path, saltenv, *a, **kwargs):
             real_path = _p(f_path)
             mani_path = _p(f_mani)
             sign_path = _p(f_sign)
-            log.debug('path=%s rpath=%s manifest=%s signature=%s',
-                path, real_path, mani_path, sign_path)
+            log.debug('path: {} | manifest: "{}" | signature: "{}"'.format(
```
Comment (jettero) [inline hubblestack/utils/signing.py]: avoid new style formatting

File: hubblestack/utils/signing.py [hunk 10]
```diff
@@ -146,8 +193,22 @@ def read_certs(*fnames):
...
+                log_level('error while reading "{}": {}'.format( fname, E))
+
+
+def stringify_cert_files(cert):
+    """this function returns a string version of cert(s) for returner"""
+    if type(cert) is list and len(cert) >= 1:
+        return ', '.join([str(c) for c in cert])
+    elif type(cert) is file:
+        return cert.name
+    else:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Unnecessary else

File: hubblestack/utils/signing.py [hunk 11]
```diff
@@ -66,6 +64,53 @@
...
+        return True
+
+    ts_1 = time()
+    # make a timedelta from the loaded ts vs now.
+    td =  ts_1 - ts_0
+    if td >= dampener_limit:
+        new_ts = time()
+        verif_log_timestamps[target] = new_ts
+        return True
+    else:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Unnecessary else

File: hubblestack/utils/signing.py [hunk 12]
```diff
@@ -124,10 +167,14 @@ def split_certs(fh):
...
                 ret = ret.encode()
                 try:
+                    log_level = log.debug
                     yield ossl.load_certificate(ossl.FILETYPE_PEM, ret)
-                except Exception as exception_object:
-                    log.debug('decoding item as certificate failed: %s; trying as PEM encoded private key',
-                        exception_object)
+                except Exception as E:
+                    status = STATUS.UNKNOWN
+                    if check_verif_timestamp(fh) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 13]
```diff
@@ -146,8 +193,22 @@ def read_certs(*fnames):
                 with open(fname, 'r') as fh:
                     for i in split_certs(fh):
                         yield i
-            except Exception as exception_object:
-                log.error('error while reading "%s": %s', fname, exception_object)
+            except Exception as E:
+                log_level = log.debug
+                if check_verif_timestamp(fname) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 14]
```diff
@@ -194,32 +256,48 @@ def __init__(self, public_crt, ca_crt):
...
             digest = i.digest('sha1')
             if digest in already:
                 continue
             already.add(digest)
             digest = digest.decode() + " " + stringify_ossl_cert(i)
-            log.debug('adding %s as a trusted certificate approver', digest)
             self.store.add_cert(i)
             self.trusted.append(digest)
+            log_level = log.debug
+            if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 15]
```diff
@@ -194,32 +256,48 @@ def __init__(self, public_crt, ca_crt):
...
                 self.trusted.append(digest)
-                log.debug('  added to verify store')
+                status = STATUS.VERIFIED 
+                log_level = log.debug
             except ossl.X509StoreContextError as exception_object:
-                log.debug('  not trustworthy: %s', exception_object)
+                # log at either log.error or log.critical according to the error code
+                status = STATUS.FAIL
+                pass
+            if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 16]
```diff
@@ -229,34 +307,44 @@ def __init__(self, public_crt, ca_crt):
...
             already.add(digest)
             digest = digest.decode() + " " + stringify_ossl_cert(i)
-            log.debug('checking to see if %s is a valid leaf cert', digest)
+            log_level = log.debug
             try:
                 ossl.X509StoreContext(self.store, i).verify_certificate()
                 status = STATUS.VERIFIED
                 self.trusted.append(digest)
-                log.debug('  marking verified')
+                if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 17]
```diff
@@ -229,34 +307,44 @@ def __init__(self, public_crt, ca_crt):
...
-                # define X509_V_ERR_CERT_HAS_EXPIRED                  10
-                # define X509_V_ERR_CRL_HAS_EXPIRED                   12
-                # XX # if code in (10,12):
-                # XX #     return # .... is this even the right idea? should we do this through conext flags?
+                if code in (10,12):
+                    # X509_V_ERR_CERT_HAS_EXPIRED                  10
+                    # X509_V_ERR_CRL_HAS_EXPIRED                   12
+                    status = STATUS.FAIL
+                # log at either log.error or log.critical according to the error code
+                if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 18]
```diff
@@ -352,44 +440,59 @@ def sign_target(fname, ofname, private_key='private.key', **kwargs): # pylint: d
...
+    log_level = log.debug
+    short_fname = fname.split('/')[-1]
     try:
         with open(sfname, 'r') as fh:
             sig,_,_ = PEM.decode(fh.read()) # also returns header and decrypted-status
     except IOError:
-        log.info('verify_signature() failed to find sfname=%s for fname=%s', sfname, fname)
+        status = STATUS.UNKNOWN
+        verif_key = ':'.join([fname, sfname])
+        if check_verif_timestamp(verif_key) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 19]
```diff
@@ -352,44 +440,59 @@ def sign_target(fname, ofname, private_key='private.key', **kwargs): # pylint: d
...
             args['padding'] = padding.PSS( mgf=padding.MGF1(hashes.SHA256()),
                 salt_length=padding.PSS.MAX_LENGTH)
             args['algorithm'] = utils.Prehashed(chosen_hash)
         try:
             pubkey.verify(**args)
+            log_level(msg.format(short_fname, fname, status, sha256sum, txt ))
             return status
         except InvalidSignature:
+            status = STATUS.FAIL
+            if check_verif_timestamp(fname) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py [hunk 20]
```diff
@@ -457,25 +563,41 @@ def verify_files(targets, mfname='MANIFEST', sfname='SIGNATURE', public_crt='pub
...
             # presumably it's signed (STATUS.VERIFIED); but perhaps it's only
             # UNKNOWN or even FAIL.
-            ret[vfname] = ret[mfname]
+            status = ret[mfname]
         else:
             # We do have a MANIFEST entry and it doesn't match: FAIL with or
             # without a matching SIGNATURE
-            ret[vfname] = STATUS.FAIL
+            status = STATUS.FAIL
+        if check_verif_timestamp(digest) == True:
```
Comment (daveed12) [inline hubblestack/utils/signing.py]: Redundant comparison to True, check_verif_timestamp() returns bool value already.

File: hubblestack/utils/signing.py (hunk 1 above)
Comment (git-gud-1) [reply hubblestack/utils/signing.py]: Got it, will format using the older style

**Comment by git-gud-1** [comment]
fixed change

**Pull request closed without merging**